src/prototype.rs

```diff
@@ -1174,21 +1174,31 @@ fn get_html_end_tag(text : &str) -> Option<&'static str> {
     static BEGIN_TAGS: &'static [&'static str; 3] = &["<script", "<pre", "<style"];
     static END_TAGS: &'static [&'static str; 3] = &["</script>", "</pre>", "</style>"];
 
-    for (beg_tag, end_tag) in BEGIN_TAGS.iter().zip(END_TAGS.iter()) {
-        if 1 + beg_tag.len() <= text.len() &&
-            text.starts_with(&beg_tag[..]) {
+    'type_1: for (beg_tag, end_tag) in BEGIN_TAGS.iter().zip(END_TAGS.iter()) {
+        if text.len() >= beg_tag.len() && text.starts_with("<") {
+            for (i, c) in beg_tag.as_bytes()[1..].iter().enumerate() {
+                if ! (&text.as_bytes()[i+1] == c || &text.as_bytes()[i+1] == &(c - 32)) {
```

Comment:
So far I haven't seen any of the ways this isn't safe, but I just noticed there's a `utils::strcasecmp()` function....any preference?

Reply:
Not a strong preference. I do think the code would be improved if the case-insensitive comparison was split out into a different function, but using `strcasecmp` is also not trivial, as it takes `&str` and here we're working with bytes. If you were going to do that, I think the way to do it is scan the keyword, detecting the end condition (eol, whitespace, `>`) _first_, then strcasecmp if that passes. To my ears, that's slightly cleaner than this, but either way seems fine.

Reply:
Resolved this one for now with a `TODO` comment.